File: mocker-front/src/app/core/service/generate.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TableModel } from '@core/domain/model/entity/table.model';
import { AbstractService } from '@core/service/abstract.service';
import { HttpMethodEnum } from '@core/domain/enum/http-method.enum';
import { DataModel } from '@core/domain/model/data.model';

/**
 * @author Do Quoc Viet
 */

@Injectable({
  providedIn: 'root'
})
export class GenerateService extends AbstractService<any> {
  override ENTITY_URL: string = 'generate';

  generateWithTable(table: TableModel): Observable<any[]> {
    return this.request<any[]>(HttpMethodEnum.POST, `${this.API_URL}/${this.ENTITY_URL}`, table);
  }

  generateBySchema(schemaId: string): Observable<DataModel> {
    return this.request<DataModel>(HttpMethodEnum.GET, `${this.API_URL}/${this.ENTITY_URL}/schema/${schemaId}`);
  }
}

File: mocker-back/src/main/java/com/mocker/service/GenerateService.java
package com.mocker.service;

import com.mocker.domain.model.entity.Table;

import java.util.List;
import java.util.Map;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

public interface GenerateService {
    List<Map<String, String>> generateWithTableId(UUID tableId);

    List<Map<String, String>> generateWithTable(Table table);

    Map<String, Map<String, Object>> generateWithSchema(UUID id);
}
